soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -3947,14 +3945,17 @@ void DrawRandoEditor(bool& open) {
                 ImGui::PushItemWidth(-FLT_MIN);
 
                 //Starting Age
-                //Disabled when Forest is set to Closed
-                bool disableRandoStartingAge = !CVar_GetS32("gRandomizeForest", 0);
-                const char* disableRandoStartingAgeText = "This option is disabled because \"Forest\" is set to \"Closed\".";
+                //Disabled when Forest is set to Closed or under very specific conditions
+                bool disableRandoStartingAge = !CVar_GetS32("gRandomizeForest", 0) || //If Forest is closed...
+                    !(CVar_GetS32("gRandomizeDoorOfTime", 0)) && //or Door of Time is Intended
+                    !(CVar_GetS32("gRandomizeShuffleOcarinas", 0)) && //and ocarinas are shuffled
+                    !(CVar_GetS32("gRandomizeLogicRules", 0)); // and logic is glitchless
```

Comment:
i feel like there are some issues with the logic here (or maybe just the comments)

`gRandomizeForest` has open/closed/closed deku, so `!CVar_GetS32("gRandomizeForest", 0)` can't be checking for closed.

door of time has open/closed/intended, so `!(CVar_GetS32("gRandomizeDoorOfTime", 0))` can't be checking for intended

i don't remember if shuffle ocarinas set to 0 means shuffle or not, so maybe `!(CVar_GetS32("gRandomizeShuffleOcarinas", 0))` is correct

`!(CVar_GetS32("gRandomizeLogicRules", 0))` should be fine for now when we just have glitchless/no logic, but eventually we'll have glitched logic in which case using `!` will break things...

Reply:
Should we checking explicitly to future proof then? It may be more verbose but I'm fine with it.
As for the comments, it is a bit tricky but the logic does work out, since `CVar_GetS32("gRandoVariable", 0)` is truthy if it doesn't return `0`:

- `gRandomizeForest` has Closed Forest set to 0, so we're looking for the inverse of the default case (truthy if it returns `0`)
- Similarly, `gRandomizeDoorOfTime` has Intended set to 0.
- `gRandomizeShuffleOcarinas` is 0 for not shuffling, so similarly we want a truthy answer if set to 1.
- Lastly, I don't mind setting `gRandomizeLogicRules` explicitly, but Glitchless is also set to 0 so it should work as well regardless.

Reply:
> `gRandomizeForest` has Closed Forest set to 0, so we're looking for the inverse of the default case (truthy if it returns `0`)

this is unintuitive: if the check is "if forest is closed" then it would make sense to me write it as `CVar_GetS32("gRandomizeForest", 0) == 0` (really all of this could be cleaned up if we didn't use a million magic numbers for cvars). in the current state, the `!` implies checking that it is *not* something (in this case not any value other than zero)

the same goes for `gRandomizeDoorOfTime`, even though `!` is logically equivalent, it is still hard to parse with the `!` and the magic numbers

for ocarina it reads decently, "if ocarinas are not shuffled" (the comment says if ocarinas *are* shuffled though)

since logic is currently boolean it seems fine to use the `!`, but since it won't always be a boolean value we should really be explicit there.

Reply:
One small thing I want to add, please use "Closed" in your comments instead of "Intended". I know 3DS calls it intended but it's confusing when we use "Closed" in our own code. The only place it's still called intended is in a 3DS enum iirc.

Reply:
![image](https://user-images.githubusercontent.com/49068535/195463657-6f513d84-8003-46b9-acd7-cea3749e9037.png)
As of last commit, the ocarina comment had been fixed, not sure why Github doesn't show it.
Otherwise, I'll go ahead and make the actual statements more clear.
We could also implement something similar to how 3d rando handled their settings by having a file full of setting enums if you think that'd be a good idea?

Reply:
having a file full of setting enums would probably be a good idea yeah, but definitely out of scope for this pr

Reply:
> having a file full of setting enums would probably be a good idea yeah, but definitely out of scope for this pr

#1761